cloudinit/cmd/clean.py

```diff
@@ -132,14 +134,34 @@ def remove_artifacts(init, remove_logs, remove_seed=False, remove_config=None):
     ):
         for conf in GEN_SSH_CONFIG_FILES:
             del_file(conf)
-    if remove_config and set(remove_config).intersection(
-        ["all", "datasource"]
-    ):
-
-        init.fetch().clean()
 
     if not os.path.isdir(init.paths.cloud_dir):
+        if remove_config and set(remove_config).intersection(
+            ["all", "datasource"]
+        ):
+            log_util.multi_log(
+                "Cache removed, datasource not found, nothing cleaned.",
+                log=LOG,
+                log_level=logging.WARNING,
+            )
         return 0  # Artifacts dir already cleaned
+
+    cache_path = os.path.join(init.paths.cloud_dir, "instance")
+    if (
+        remove_config
+        and set(remove_config).intersection(["all", "datasource"])
```

Comment:
`set(remove_config).intersection(["all", "datasource"])` is computed and parsed by readers two times (here and in the previous branch), do you think it would be better to store it in a variable `clean_datasource`, or something in those lines?